src/robot/libraries/Collections.py

```diff
@@ -1277,17 +1326,25 @@ def _report_error(default_message, message, values=False):
 
 class Normalizer:
 
-    def __init__(self, ignore_case=False, ignore_order=False, ignore_keys=None):
-        self.ignore_case = ignore_case
+    def __init__(
+        self,
+        ignore_case: IgnoreCase = False,
+        ignore_order: bool = False,
+        ignore_keys: "Iterable[Hashable] | str | None" = None,
+    ):
         if isinstance(ignore_case, str):
             self.ignore_key_case = ignore_case.upper() not in ("VALUE", "VALUES")
             self.ignore_value_case = ignore_case.upper() not in ("KEY", "KEYS")
```

Comment:
`upper()` calls aren't needed anymore. Preserving them in case someone calls these keywords programmatically is fine, though.